pkgs/tools/typesetting/sile/default.nix

```diff
@@ -9,13 +9,15 @@
 , icu
 , fontconfig
 , lua
+# Use this to override the lua environment used by sile
+, luaEnv ? null
```

Comment:
For the sake of the default packages yes I think you can (as we were doing up until this PR). I think the issue is that this method makes it very difficult to modify the lua downstream in the Nix Flake for the project ... we can't add new Lua dependencies in the Flake as only the modifications since the last stable release. Background here: https://github.com/sile-typesetter/sile/issues/1388#issuecomment-1143413539

Reply:
What we need downstream is some way to add dependencies (and occasionally remove dependencies) in the downstream flake.nix while still importing most of the dependency and build configuration from this package.

Reply:
you could filter the deps by name to remove specific ones and do some fancy stuff but wouldn't it be easier overall to copy/paste the definition in the flake ?

Reply:
Adding things is easy. Removing is more complicated. It's probably the easiest to just copy luaEnv and then change how you'd like.

Reply:
If there are not strong objections to this PR, and considering @alerque is a bit stressed to see this change in the next `nix flake update`, I'd like to merge this.

Reply:
th proposed change makes the expression harder to read for what seems to be not a great reason. You should be able to override the luaEnv by adding a new one to the buildInputs, just be careful in what order you specify it (new one should come first). Worst case you can override the current expression and just override fully the bits (buildInputs for instance) you are interested in

Reply:
> new one should come first

Indeed adding the overridden `luaEnv` as the first `buildInput` works, thanks for that. I still think though that this proposal makes things much more elegant, (in general, we'd like to avoid writing things twice in the flake expression and in the expression here). Having eventually two `luaEnv` inputs seems not ideal, but I guess not harmful. Perhaps when [__structuredAttrs](https://github.com/NixOS/nixpkgs/pull/175649) will be supported in nixpkgs, we'd be able to override such inputs easily.